data-prepper-core/src/test/java/org/opensearch/dataprepper/pipeline/PipelineConnectorTest.java

```diff
@@ -78,6 +184,42 @@ public void testOutputSuccess() throws Exception {
         verify(buffer).write(eq(RECORD), anyInt());
     }
 
+    @Test
+    public void testEventBufferOutputSuccess() throws Exception {
+        eut.start(eventBuffer);
+
+        eut.output(eventRecordList);
+
+	Map.Entry<Collection<Record<Event>>, CheckpointState> ent = eventBuffer.read(1);
+	ArrayList<Record<Event>> records = new ArrayList<>(ent.getKey());
+	// Make sure the records are different
+	Assertions.assertNotEquals(eventRecordList.get(0),  records.get(0));
+	// Make sure the events are different
+	Event event1 = eventRecordList.get(0).getData();
+	Event event2 = records.get(0).getData();
+	Assertions.assertNotEquals(event1, event2);
```

Comment:
It would be clearer (and more accurate) to assert these are not the same instance.

```
assertThat(event1, not(sameInstance(event2));
```